AK: TestCircularDuplexStream fails to compile due to possibly uninitialized variable

Trying to build serenity is failing on my system on AK/Tests/TestCircularDuplexStream.cpp due to a possibly uninitialized use of the `byte` variable. Here's my full ninja output:

```[0/2] Re-checking globbed directories...
[141/2434] Building CXX object AK/Test...eam.dir/TestCircularDuplexStream.cpp.o
FAILED: AK/Tests/CMakeFiles/TestCircularDuplexStream.dir/TestCircularDuplexStream.cpp.o 
/usr/bin/c++  -I../Libraries -I../. -O3 -pipe -march=native -Wno-unknown-warning-option -Wall -Wextra -Werror -Wmissing-declarations -std=c++2a -fdiagnostics-color=always -ftls-model=initial-exec -fconcepts -MD -MT AK/Tests/CMakeFiles/TestCircularDuplexStream.dir/TestCircularDuplexStream.cpp.o -MF AK/Tests/CMakeFiles/TestCircularDuplexStream.dir/TestCircularDuplexStream.cpp.o.d -o AK/Tests/CMakeFiles/TestCircularDuplexStream.dir/TestCircularDuplexStream.cpp.o -c ../AK/Tests/TestCircularDuplexStream.cpp
In file included from ../AK/Tests/TestCircularDuplexStream.cpp:27:
.././AK/TestSuite.h: In function ‘void __test_of_by_one()’:
.././AK/TestSuite.h:284:9: error: ‘byte’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  284 |         if (lhs != rhs)                                                                                                                                                                \
      |         ^~
../AK/Tests/TestCircularDuplexStream.cpp:102:12: note: ‘byte’ was declared here
  102 |         u8 byte;
      |            ^~~~
In file included from ../AK/Tests/TestCircularDuplexStream.cpp:27:
.././AK/TestSuite.h: In function ‘void __test_works_like_a_queue()’:
.././AK/TestSuite.h:283:14: error: ‘byte’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  283 |         auto rhs = (b);                                                                                                                                                                \
      |              ^~~
../AK/Tests/TestCircularDuplexStream.cpp:44:12: note: ‘byte’ was declared here
   44 |         u8 byte;
      |            ^~~~
In file included from ../AK/Tests/TestCircularDuplexStream.cpp:27:
.././AK/TestSuite.h: In function ‘void __test_overwritting_is_well_defined()’:
.././AK/TestSuite.h:284:13: error: ‘byte’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  284 |         if (lhs != rhs)                                                                                                                                                                \
      |             ^~~
../AK/Tests/TestCircularDuplexStream.cpp:74:12: note: ‘byte’ was declared here
   74 |         u8 byte;
      |            ^~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
[146/2434] Building CXX object Librari...akeFiles/RegexLibC.dir/RegexLibC.cpp.o
ninja: build stopped: subcommand failed.```

I am on Gentoo, using ninja 1.10.2 as a make replacement as per the build instructions.

--- a/AK/CircularDuplexStream.h
+++ b/AK/CircularDuplexStream.h
@@ -54,7 +54,8 @@ public:
             return false;
         }
 
-        write(bytes);
+        const auto nwritten = write(bytes);
+        ASSERT(nwritten == bytes.size());
         return true;
     }
 

--- a/AK/Tests/TestCircularDuplexStream.cpp
+++ b/AK/Tests/TestCircularDuplexStream.cpp
@@ -60,12 +60,11 @@ TEST_CASE(overwritting_is_well_defined)
     for (size_t idx = 0; idx < capacity; ++idx)
         stream << static_cast<u8>(idx % 256);
 
-    u8 bytes[half_capacity];
-
-    stream >> Bytes { bytes, sizeof(bytes) };
+    Array<u8, half_capacity> buffer;
+    stream >> buffer;
 
     for (size_t idx = 0; idx < half_capacity; ++idx)
-        EXPECT_EQ(bytes[idx], idx % 256);
+        EXPECT_EQ(buffer[idx], idx % 256);
 
     for (size_t idx = 0; idx < half_capacity; ++idx)
         stream << static_cast<u8>(idx % 256);
@@ -81,33 +80,6 @@ TEST_CASE(overwritting_is_well_defined)
     }
 
     EXPECT(stream.eof());
-}
-
-TEST_CASE(of_by_one)
-{
-    constexpr size_t half_capacity = 32;
-    constexpr size_t capacity = half_capacity * 2;
-
-    CircularDuplexStream<capacity> stream;
-
-    for (size_t idx = 0; idx < half_capacity; ++idx)
-        stream << static_cast<u8>(0);
-
-    for (size_t idx = 0; idx < half_capacity; ++idx)
-        stream << static_cast<u8>(1);
-
-    stream.discard_or_error(capacity);
-
-    for (size_t idx = 0; idx < capacity; ++idx) {
-        u8 byte;
-        stream.read({ &byte, sizeof(byte) }, capacity);
-        stream << byte;
-
-        if (idx < half_capacity)
-            EXPECT_EQ(byte, 0);
-        else
-            EXPECT_EQ(byte, 1);
-    }
 }
 
 TEST_MAIN(CircularDuplexStream)

